src/html-media-source.js

```diff
@@ -180,6 +180,24 @@ export default class HtmlMediaSource extends videojs.EventTarget {
       }
     });
 
+    this.on('sourceended', (event) => {
+      let duration = this.duration;
+
+      for (let i = 0; i < this.sourceBuffers.length; i++) {
+        let sourcebuffer = this.sourceBuffers[i];
+        let cues = sourcebuffer.metadataTrack_.cues;
+
+        if (cues) {
+          if (isNaN(duration) || Math.abs(duration) === Infinity) {
```

Comment:
The duration logic should be done moved to before the for loop since duration won't change with each duration. Also could just call the function if you export from add-text-track-data.js as mentioned in my other comment.